### ISSUE TITLE
Ubuntu in Docker container produces Initialization Segmentation Fault 

### ISSUE THREAD

**Issue by MatthewZalesak** [JuliaPy/pyjulia#119]
This issue seems like it might be similar to issue #92.  I am starting with a default Docker container for Ubuntu.  I run through the steps of a basic installation which roughly follows the directions on the project's README.  However, when I initialize the Julia object in Python I get a segmentation fault.

Code that produces the error:

```
sudo docker run -it ubuntu
apt-get update
apt-get install git python python-pip curl

cd
curl https://julialang-s3.julialang.org/bin/linux/x64/0.6/julia-0.6.0-linux-x86_64.tar.gz -o julia.tar.gz
tar xvf julia.tar.gz
mv julia-903644385b /opt/
ln -s /opt/julia-903644385b/bin/julia /bin/julia

julia
Pkg.add("PyCall") # OUTPUT: PyCall is using python at /usr/bin/python, libpython = libpython2.7
Pkg.installed("PyCall") # OUTPUT: v"1.14.0"
exit()

pip2 install git+https://github.com/JuliaPy/pyjulia
python
import julia
j = julia.Julia()
```
The expected output from this would be no error message from Python.  However, the above produces the error message:

```
InitError(:PyCall, PyError (ccall(@pysym(:PyImport_ImportModule), PyPtr, (Cstring,), name)) <type 'exceptions.SystemError'>
SystemError('dynamic module not initialized properly',)

signal (11): Segmentation fault
while loading no file, in expression starting on line 0
PyObject_Call at /usr/lib/x86_64-linux-gnu/libpython2.7.so (unknown line)
macro expansion at /root/.julia/v0.6/PyCall/src/exception.jl:78 [inlined]
#_pycall#66 at /root/.julia/v0.6/PyCall/src/PyCall.jl:596
jl_call_fptr_internal at /home/centos/buildbot/slave/package_tarball64/build/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /home/centos/buildbot/slave/package_tarball64/build/src/julia_internal.h:358 [inlined]
jl_apply_generic at /home/centos/buildbot/slave/package_tarball64/build/src/gf.c:1933
jl_apply at /home/centos/buildbot/slave/package_tarball64/build/src/julia.h:1424 [inlined]
jl_f__apply at /home/centos/buildbot/slave/package_tarball64/build/src/builtins.c:426
_pycall at /root/.julia/v0.6/PyCall/src/PyCall.jl:584
jl_call_fptr_internal at /home/centos/buildbot/slave/package_tarball64/build/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /home/centos/buildbot/slave/package_tarball64/build/src/julia_internal.h:358 [inlined]
jl_apply_generic at /home/centos/buildbot/slave/package_tarball64/build/src/gf.c:1933
jl_apply at /home/centos/buildbot/slave/package_tarball64/build/src/julia.h:1424 [inlined]
jl_f__apply at /home/centos/buildbot/slave/package_tarball64/build/src/builtins.c:426
#pycall#69 at /root/.julia/v0.6/PyCall/src/PyCall.jl:615
jl_call_fptr_internal at /home/centos/buildbot/slave/package_tarball64/build/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /home/centos/buildbot/slave/package_tarball64/build/src/julia_internal.h:358 [inlined]
jl_apply_generic at /home/centos/buildbot/slave/package_tarball64/build/src/gf.c:1933
jl_apply at /home/centos/buildbot/slave/package_tarball64/build/src/julia.h:1424 [inlined]
jl_invoke at /home/centos/buildbot/slave/package_tarball64/build/src/gf.c:51
show at /root/.julia/v0.6/PyCall/src/exception.jl:39
unknown function (ip: 0x7fd9212a5106)
jl_call_fptr_internal at /home/centos/buildbot/slave/package_tarball64/build/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /home/centos/buildbot/slave/package_tarball64/build/src/julia_internal.h:358 [inlined]
jl_apply_generic at /home/centos/buildbot/slave/package_tarball64/build/src/gf.c:1933
show_default at ./show.jl:140
unknown function (ip: 0x7fd9212a2d86)
jl_call_fptr_internal at /home/centos/buildbot/slave/package_tarball64/build/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /home/centos/buildbot/slave/package_tarball64/build/src/julia_internal.h:358 [inlined]
jl_apply_generic at /home/centos/buildbot/slave/package_tarball64/build/src/gf.c:1933
show at ./show.jl:125
unknown function (ip: 0x7fd9212a2426)
jl_call_fptr_internal at /home/centos/buildbot/slave/package_tarball64/build/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /home/centos/buildbot/slave/package_tarball64/build/src/julia_internal.h:358 [inlined]
jl_apply_generic at /home/centos/buildbot/slave/package_tarball64/build/src/gf.c:1933
jl_apply at /home/centos/buildbot/slave/package_tarball64/build/src/julia.h:1424 [inlined]
jl_call2 at /home/centos/buildbot/slave/package_tarball64/build/src/jlapi.c:204
ffi_call_unix64 at /usr/lib/x86_64-linux-gnu/libffi.so.6 (unknown line)
ffi_call at /usr/lib/x86_64-linux-gnu/libffi.so.6 (unknown line)
_ctypes_callproc at /usr/lib/python2.7/lib-dynload/_ctypes.x86_64-linux-gnu.so (unknown line)
unknown function (ip: 0x7fd94372dd81)
PyObject_Call at python (unknown line)
PyEval_EvalFrameEx at python (unknown line)
PyEval_EvalCodeEx at python (unknown line)
PyEval_EvalFrameEx at python (unknown line)
PyEval_EvalCodeEx at python (unknown line)
PyEval_EvalFrameEx at python (unknown line)
PyEval_EvalCodeEx at python (unknown line)
unknown function (ip: 0x4de6fd)
PyObject_Call at python (unknown line)
unknown function (ip: 0x4f492d)
PyObject_Call at python (unknown line)
unknown function (ip: 0x4f46a6)
unknown function (ip: 0x4b670b)
PyObject_Call at python (unknown line)
PyEval_EvalFrameEx at python (unknown line)
PyEval_EvalCodeEx at python (unknown line)
PyEval_EvalCode at python (unknown line)
unknown function (ip: 0x4f1dee)
PyRun_InteractiveOneFlags at python (unknown line)
PyRun_InteractiveLoopFlags at python (unknown line)
unknown function (ip: 0x42e889)
Py_Main at python (unknown line)
__libc_start_main at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
_start at python (unknown line)
Allocations: 1708595 (Pool: 1707396; Big: 1199); GC: 1
Segmentation fault (core dumped)
```
Running Pkg.rm("PyCall"), Pkg.add("PyCall") to force a rebuild as the first comment of #92 suggested does not change the error message.

**Comment by davidbuniat** [comment]
Same here

**Comment by MatthewZalesak** [comment]
I believe I have found the solution to this problem.

The problem seems to stem from the way that Julia installs packages.  Whenever a package is added with Pkg.add() a git repository is added to the local .julia folder.  However, this is not always the most recent version of the package.  In particular it seems that the version of PyCall that Julia installs is not compatible with pyjulia.  I ran the following code to update the PyCall package and this problem was resolved.

```
cd ~/.julia/v0.6/PyCall
git pull origin master
git checkout master
julia
Pkg.build("PyCall")
exit()
```

Julia seems to install PyCall commit f863d82b34969017432be784954029575b2e3e61 from Jul 3, 2017 while the most recent version is commit 0550f890c86878375f74821a459da41e7cc943f6 from Aug 21, 2017.
